tests/testUpdate.py

```diff
@@ -0,0 +1,64 @@
+import os
+import shutil
+import unittest
+from base import *
+
+
+class TestUpdate(TestZstash):
+    # x = on, no mark = off, b = both on and off tested
+    # option | Update | UpdateDryRun | TestZstash.add_files
+    # --hpss    |x|x|x|
+    # --dry-run | |x| |
+    # --keep    | | |b|
+    # -v        | | |b|
+   
+    def helperUpdate(self, test_name, hpss_path, zstash_dir=ZSTASH_DIR):
+        self.hpss_path = hpss_path
+        use_hpss = self.setupDirs(test_name)
+        self.create(use_hpss, zstash_dir)
+        print_starred('Running update on the newly created directory, nothing should happen')
+        self.assertWorkspace()
+        os.chdir(self.test_dir)
+        cmd = '{}zstash update -v --hpss={}'.format(zstash_dir, self.hpss_path)
+        output, err = run_cmd(cmd)
+        os.chdir(TOP_LEVEL)
+        self.check_strings(cmd, output + err, ['Nothing to update'], ['ERROR'])
+
+    def helperUpdateDryRun(self, test_name, hpss_path, zstash_dir=ZSTASH_DIR):
+        self.hpss_path = hpss_path
+        use_hpss = self.setupDirs(test_name)
+        self.create(use_hpss, zstash_dir)
+        print_starred('Testing update with an actual change')
+        self.assertWorkspace()
+        if not os.path.exists('{}/dir2'.format(self.test_dir)):
+            os.mkdir('{}/dir2'.format(self.test_dir))
+        write_file('{}/dir2/file2.txt'.format(self.test_dir), 'file2 stuff')
+        write_file('{}/dir/file1.txt'.format(self.test_dir), 'file1 stuff with changes')
+
+        os.chdir(self.test_dir)
+        cmd = '{}zstash update --dry-run --hpss={}'.format(zstash_dir, self.hpss_path)
+        output, err = run_cmd(cmd)
+        os.chdir(TOP_LEVEL)
+        expected_present = ['List of files to be updated',
+             'dir/file1.txt',
+             'dir2/file2.txt']
+        expected_absent = ['ERROR', 'file0', # Make sure none of the old files are moved.
+            'file_empty', 'empty_dir']
+        self.check_strings(cmd, output + err, expected_present, expected_absent)
```

Comment:
Is this test sufficient to show that `update` didn't actually run, merely printing what it would have done?